highway-economics.tex

```diff
@@ -25,9 +25,9 @@
 
 \title{Economics of the CasperLabs Highway Protocol}
```

Comment:
I agree that finalizing the switch block is very important. To me, the problem can be interpreted both ways:

Pessimistic: We have to make sure that validators receive rewards for finalizing the switch block, because otherwise they may not be incentivized to keep the network from stalling.

Optimistic: It's in the validators best interest for the switch block to be finalized, which they would even do it for free, because the external cost of the network stalling chronically at switch blocks is simply too much for them. Maybe, finalizing 1 block for free every round would not be too big of a deal for them.

I would argue for the latter only if the workaround to reward switch blocks prove to add too much complexity. Otherwise, I'm on your side: finalizing switch blocks should be rewarded either with seigniorage just as other blocks, or we treat them as an exception and reward them through some indirect means.

Reply:
> if the workaround to reward switch blocks prove to add too much complexity.

I even think it might be theoretically impossible without opening the door to some resource exhaustion attack again (forcing everyone to download a huge amount of data from the previous era).

So if we go with the optimistic approach, what's the argument for having fine-grained round rewards in the first place? Couldn't you argue that finalization is in the validators' best interest anyway, so you can pay them all the same, independent of how they behaved?

I guess the answer is that you could save a lot of cost by not participating at all, but only negligible cost by not participating in finalizing the switch block?

Reply:
> I guess the answer is that you could save a lot of cost by not participating at all, but only negligible cost by not participating in finalizing the switch block?

Right.

> what's the argument for having fine-grained round rewards in the first place? Couldn't you argue that finalization is in the validators' best interest anyway, so you can pay them all the same, independent of how they behaved?

I don't think the optimistic approach can be reduced to what you describe. Validators would still be rewarded according to their performance and behavior 99.9% of the time. The argument is that the 0.1% difference wouldn't make a lot of difference, and can be neglected. They have a very strong indirect incentive to finalize the switch block, because their revenue in the subsequent rounds depend on it.